### PR TITLE
build: Refactor to consolidate the rook.io/v1 package

### PR DESCRIPTION
**Pull request by travisn**
**Description of your changes:**
The rook.io/v1 package was only an internal implementation detail and does not have any CRDs that rely on it. The CRD deserialization will handle the conversion in internal types. This consolidation gives more flexibility for the storage providers to implement exactly what is needed for their storage provider instead of forcing to use the same types and risk affecting another storage provider.

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

**Review by leseb — APPROVED** [review]
Makes sense!

File: pkg/apis/ceph.rook.io/v1/labels.go [hunk 1]
```diff
@@ -16,41 +16,73 @@ limitations under the License.
...
+			ret[k] = v
+		}
+	}
+	for k, v := range with {
+		if _, ok := ret[k]; !ok {
+			ret[k] = v
+		}
+	}
+	return ret
+}
```
Comment (BlaineEXE) [inline pkg/apis/ceph.rook.io/v1/labels.go:53]: `Labels` is such a low-level type. If we have shared functionality that goes along with it like this, I don't understand why we don't keep it in the `rook.io/v1` package for other backends to reuse if they choose.

I definitely agree that having aggregate types in `rook.io/v1` is sometimes a challenge, but I think there is real use to have the low-level types like this there.

Comment (travisn) [reply pkg/apis/ceph.rook.io/v1/labels.go]: Ok after a bit more experimentation, I've moved the annotations and labels to the `rook.io` package. The version subdirectories such as `v1` just don't seem like the right place for these basic types. I could consider a `rook.io/core`, `rook.io/common`, or similar package, but directly in `rook.io` seems just as appropriate. Thoughts?

Comment (BlaineEXE) [reply pkg/apis/ceph.rook.io/v1/labels.go]: I think that seems pretty reasonable. 👍

**Pull request merged**